bundles/http/Citrix/00000019-0000-0000-0000-000000000000/47a6eb95-6f9d-44df-8be7-2322ada5770c.js

```diff
@@ -0,0 +1,849 @@
+            documentsLists.requisitions.push(json[j].approvableUniqueName)
+        }
+        if (json[j].restResourceName == 'invoices') {
+            documentsLists.invoices.push(json[j].approvableUniqueName)
+        }
+    }
+    return documentsLists
+}   
+
+function fetchChangesGetDocumentsAndGetLastId(client, documentsLists) {
+    let noOfPages = 1
+   
+    for (let i = 0; i < noOfPages; i++) {  
+        let responseData = client.fetchSync('/changes?needTotal=true&' + queryParameters(i) + '&lastChangeId=' + lastChangeId, {headers: {'apikey':apiKey}})
+        if (responseData.ok) { 
+            let json = responseData.jsonSync()
+            if (notEmptyArray(json)) {
+                if (i == 0) { //calculating number of pages for pagination
+				    noOfPages = calculateNumberOfPages(+responseData.headers.get('X-Total-Count'))
+			    }
+                if (documentsLists.newChangeId < +json[json.length - 1].changeSequenceId) { //getting the  lastChangeSequenceId for incremental sync from page
+				    documentsLists.newChangeId = +json[json.length - 1].changeSequenceId
+                }
+                documentsLists = parseChangesGetDocIds(json, documentsLists) //creating list on requisition and invoices
+		    }
+	    } else {
+            console.log( 'Warning: unable to get changes: ' + responseData.statusText)
+        }
+    }
+    return documentsLists
+}   
+
+function syncAriba(dataStore, client, context, incrementalSync) 
+{
+	console.log('Ariba sync started')
+        
+    let documentsLists  = { //creating lists od rq and invc ids
+        requisitions: [],
+        invoices: [],
+        newChangeId: lastChangeId
+       }
+    
+    if (incrementalSync) {     
+        if (context.lastChangeSequenceId > lastChangeId) {     // reading stored lastChangeSequenceId for incremental sync
+            lastChangeId = context.lastChangeSequenceId
+        }
+    }
+    else {
+        documentsLists = fetchApprovablesAndGetDocuments(client, documentsLists) //geting documents id, this endpoint is used only in full sync hold documents for open approvals infinitely
+    }
+            
+    documentsLists = fetchChangesGetDocumentsAndGetLastId(client, documentsLists) //geting documents id, this endpoint is used for full and incr, hold all data only for last 90 days 
+
+    storeRequisition(dataStore, client, documentsLists.requisitions) //storing requisitions and resolve group approvables
+          
+    if (documentsLists.newChangeId > lastChangeId ) {context.lastChangeSequenceId = documentsLists.newChangeId}     //saving  lastChangeSequenceId for incremental sync into context
+    
+    console.log('Max Change ID ' + context.lastChangeSequenceId)
+	console.log('Total number of requisitions ' + documentsLists.requisitions.length)
+	console.log('Total number of invoices ' + documentsLists.invoices.length)
+    console.log('Sync finished')
+    return
+}
+
+function setIntegrationParametersGlobal(integrationParameters){
```

Comment:
not very nice solution. 
Why can't you just take it from integrationParameters where it is needed?

Reply:
Changed, passing parameters to all functions...